unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/Utils/DestroyParticlesOnFinish.cs

```diff
@@ -25,6 +28,16 @@ public void FixedUpdate()
             return;
 
         if (particles != null && !particles.IsAlive())
+        {
+            if (Finished != null)
+            {
+                Finished.Invoke();
+                Finished = null;
+                gameObject.SetActive(false);
+                return;
```

Comment:
I see an issue with pattern, it's ofuscating logic, `DestroyParticlesOnFinish` now is somehow aware of the existance of the pool.

There are a couple of options to solve this. I think the best one is to have a new script `ReleaseParticlesOnFinish`. This would require the `PoolableObject` to be injected (probably via an Initialize method). A side benefit is avoiding the `Finished` event, with a reference to the poolable object, the script can release it itself.